services/Service.go

```diff
@@ -54,3 +55,33 @@ func (s *Service) call(ctx context.Context, methodName string, args ...interface
 
 	return resArr, nil
 }
+
+// retrieveInfo retrieves information for the given object
+func (s *Service) retrieveInfo(ctx context.Context, methodName string, userID int) (*etree.Document, error) {
```

Comment:
The `userID` parameter should be probably called differently since this method will be used by (maybe) all resource classes.